puppet/chassis.rb

```diff
@@ -134,6 +134,15 @@ def self.normalize_config(config)
 					config["mapped_paths"][path] = config["mapped_paths"]["base"]
 				end
 			end
+
+			machine_name = '' unless config["machine_name"]
+
+			# If we don't have a machine name in the config generate one from the directory structure.
+			if machine_name == ''
+				pwd = Dir.pwd.downcase
+				# Replace all non word characters with underscores then trim underscores.
```

Comment:
FYI then machine name can actually contain spaces i.e. `rob is a champion` is a valid name.